src/merge_iterator.rs

```diff
@@ -254,28 +232,36 @@ impl KeyValueIterator for MergeIterator<'_> {
             if peeked_entry.seq > current_kv.seq {
                 current_kv = peeked_entry.clone();
             }
-            self.advance().await?;
+            self.take_and_advance().await?;
         }
         Ok(Some(current_kv))
     }
 
-    async fn seek(&mut self, next_key: &[u8]) -> Result<(), SlateDBError> {
-        let mut seek_futures = VecDeque::new();
-        if let Some(iterator) = self.current.take() {
-            seek_futures.push_back(iterator.seek(next_key))
-        }
-
-        for iterator in self.iterators.drain() {
-            seek_futures.push_back(iterator.0.seek(next_key));
-        }
+    fn peek(&self) -> Option<&RowEntry> {
+        self.peek()
```

Comment:
will this have recursive call? how about renaming the inner one like `peek_inner()`?